fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java

```diff
@@ -2044,16 +2059,43 @@ private BigDecimal getDisbursementAmount(final LoanApplicationTerms loanApplicat
         return interestCharges;
     }
 
+    private BigDecimal calcUnutilizedAmount(final LoanApplicationTerms loanApplicationTerms, LocalDate untilDate,
+            boolean inRevolvingPeriodOnly) {
+        BigDecimal disbursedAmount = getDisbursedAmount(loanApplicationTerms, untilDate);
+        Money loanPrincipal = loanApplicationTerms.getApprovedPrincipal() != null
+                && loanApplicationTerms.getApprovedPrincipal().isGreaterThanZero() ? loanApplicationTerms.getApprovedPrincipal()
+                        : loanApplicationTerms.getOriginalProposedPrincipal();
+        BigDecimal unutilizedAmount = loanPrincipal.getAmount().subtract(disbursedAmount);
+        ;
+
+        // If revolving apply charge to revolving only
+        LocalDate revolvingPeriodEndDate = loanApplicationTerms.getRevolvingPeriodEndDate();
+        LocalDate revolvingPeriodStartDate = loanApplicationTerms.getRevolvingPeriodStartDate();
+        if (inRevolvingPeriodOnly && revolvingPeriodStartDate != null && revolvingPeriodEndDate != null) {
+            if ((untilDate.isBefore(revolvingPeriodEndDate) || untilDate.isEqual(revolvingPeriodEndDate))
+                    && (untilDate.isAfter(revolvingPeriodStartDate) || untilDate.isEqual(revolvingPeriodStartDate))) {
+                unutilizedAmount = loanPrincipal.getAmount().subtract(disbursedAmount);
```

Comment:
no consideration for the repaid principal for revolving credit products

Reply:
kk